### PR TITLE
Create customer repository class and spec

### LINKED ISSUE #30: Iteration 3.5 Customer Repository
CustomerRepository
Customers represent a person who’s made one or more purchases in our system.

The CustomerRepository is responsible for holding and searching our Customers instances. It offers the following methods:

all - returns an array of all known Customers instances
find_by_id - returns either nil or an instance of Customer with a matching ID
find_all_by_first_name - returns either [] or one or more matches which have a first name matching the substring fragment supplied
find_all_by_last_name - returns either [] or one or more matches which have a last name matching the substring fragment supplied
create(attributes) - create a new Customer instance with the provided attributes. The new Customer’s id should be the current highest Customer id plus 1.
update(id, attribute) - update the Customer instance with the corresponding id with the provided attributes. Only the customer’s first_name and last_name can be updated. This method will also change the customer’s updated_at attribute to the current time.
delete(id) - delete the Customer instance with the corresponding id
The data can be found in data/customers.csv so the instance is created and used like this:

```rb
sales_engine = SalesEngine.from_csv(:customers => "./data/customers.csv")
customer = sales_engine.customers.find_by_id(6)
# => <Customer>
```

2points

### PR DESCRIPTION
**Pull request by NoahZinter**
Created customer repository class/spec

Review:
Does this also need to be modified to comply with mock_data changes? 

-[x] Test Pass
-[x] Ready to merge

closes #30

### CODE REVIEW

File: lib/customer_repository.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  def create(attributes)
+    new_customer = Customer.new(attributes)
+    new_customer.update_id(find_max_id + 1)
+    @customers << new_customer
+  end
+
+  def update(id, attributes)
+    customer = find_by_id(id)
+    unless customer.nil?
```
[the remark this answers is not in the document]
Comment (mkrumholz) [reply lib/customer_repository.rb:46]: @NoahZinter I think this is prompting you to use 1-line syntax here like: 
`customer = find_by_id(id) unless customer.nil?`

**Review by mkrumholz — APPROVED** [review]
Looks good now that changes are merged from InvoiceItem!

File: lib/customer.rb [hunk 2]
```diff
@@ -12,4 +12,23 @@ def initialize(details)
...
+  end
+
+  def update_last_name(name)
+    return false if name.nil?
+    @last_name = name
+  end
+
+  def update_time
+    @updated_at = Time.now
+  end
```
Comment (sami-p) [inline lib/customer.rb:33]: Nice!!

File: spec/customer_repository_spec.rb [hunk 3]
```diff
@@ -0,0 +1,222 @@
+require 'rspec'
+require 'csv'
+require './data/mockable'
+require './lib/file_io'
+require './lib/customer_repository'
```
Comment (sami-p) [inline spec/customer_repository_spec.rb:5]: if we continue to organize alphabetically, we'll have to switch these two at some point.

**Review by sami-p — APPROVED** [review]

**Pull request merged**